src/ast/passes/resource_analyser.cpp

```diff
@@ -383,29 +371,19 @@ void ResourceAnalyser::visit(Call &call)
       resources_.max_map_key_size = std::max(resources_.max_map_key_size,
                                              map_key_size);
     }
-  } else if (call.func == "has_key") {
+  }
+  // getAssignRewriteFuncs includes the above three functions, which is why this
+  // is an else/if
```

Comment:
This is a bit confusing, maybe it could be structured as:

```
if (getAssignRewriteFuncs().contains(call.func) || ...) {
   if (call.func == "lhist" || ...) {
       ...
   } else {
       ..
   }
}
```